### PR TITLE
Move org.eclipse.ui.examples.filesystem to eclipse.platform.resources

### PR DESCRIPTION
**Pull request by pzi-dsa**
As suggested in eclipse-platform/www.eclipse.org-eclipse#1, I'd like to move the org.eclipse.ui.examples.filesystem project from [www.eclipse.org-eclipse](https://github.com/eclipse-platform/www.eclipse.org-eclipse) to this repository.

### CODE REVIEW

File: examples/pom.xml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (vogella) [inline examples/pom.xml:1]: Do we need this file? Looks like resources has pomless enabled.

**Review by vogella — CHANGES_REQUESTED** [review]
Please enhance the commit message, I think you can reuse the description from the PR.

File: examples/pom.xml (hunk 1 above)
Comment (pzi-dsa) [reply examples/pom.xml:1]: As far as I understand, the pomless aggregator only works for a [fixed](https://github.com/eclipse/tycho/blob/master/tycho-extras/tycho-pomless/src/main/java/org/eclipse/tycho/pomless/TychoAggregatorMapping.java) set of directories.

Both the _bundles_ and _tests_ directory are within this set but _examples_ is not. Which is why the pom file is not necessary for those cases.

I've also tried specifying the artifact directly. (i.e. via \<module\>example/org.eclipse.ui.examples.filesystem\</module\>)
But in this case Tycho still requires a pom file in the _examples_ directory. Here's the error:

> Non-readable POM C:\[...]\eclipse.platform.resources\examples\org.eclipse.ui.examples.filesystem\.polyglot.META-INF: No parent pom file found in C:\[...]\eclipse.platform.resources\examples

**Review by vogella — CHANGES_REQUESTED** [review]
Patrick, the foundation requested that all Eclipse projects migrate to EPL 2.0. Could you update the copy-right header of the files with the usage of the EPL.  I think it is fine just to use the same copyright for all files, if that makes your life simpler.

Sorry for this request, I was the one who did most of the EPL 1.0 EPL 2.0 work a while ago, so I know how annoying this task is. For a valid copyright header, please check any other file in the repo.

File: examples/pom.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+<!--
+  Copyright (c) 2013, 2015 Eclipse Foundation and others.
```
[the remark this answers is not in the document]
Comment (pzi-dsa) [reply examples/pom.xml:3]: There are some classes without a license header (e.g. CollapseZipAction). Should I add the EPL 2.0 header there as well or leave them as is?

[the remark this answers is not in the document]
Comment (vogella) [reply examples/pom.xml:3]: Please add also the licence header.

[the remark this answers is not in the document]
Comment (pzi-dsa) [reply examples/pom.xml]: Updated.

**Comment by vogella** [comment]
LGTM, lets see that the CI thinks of this. This repo has some random tests errors (see issues) so tests errors are most likely not related.

**Comment by pzi-dsa** [comment]
I have removed almost all compiler warnings and the only ones remaining so far are due to the usage of org.eclipse.core.internal.filesystem.Messages/Policy.

I can't replace them with the Eclipse logger because they are used to construct CoreExceptions. Does anyone know of a good alternative? Would it be alright to simply ignore those warnings?

**Comment by vogella** [comment]
Checking in now...

**Comment by pzi-dsa** [comment]
> Can you set he project encoding (right mouse click on project and Properties) to UTC-8 and text file delimitor to Unix?

Done.

**Comment by vogella** [comment]
> > Can you set he project encoding (right mouse click on project and Properties) to UTC-8 and text file delimitor to Unix?
> 
> Done.

Cool, lets see if that makes CI happy.

**Comment by vogella** [comment]
The build error looks unrelated to me, lets wait a bit maybe this is just a temporay hick-up. I check again tomorrow.

**Comment by pzi-dsa** [comment]
> error: warnings found and -failOnWarning specified

I assume that's the problem, because the example (ZipFileStore, to be specific) is using some internal Eclipse classes for "logging" purposes.

I have already resolved most of the remaining warnings, but this one is something where I haven't found a good alternative yet.
Do you have any suggestions?

**Comment by vogella** [comment]
> > error: warnings found and -failOnWarning specified
> 
> I assume that's the problem, because the example (ZipFileStore, to be specific) is using some internal Eclipse classes for "logging" purposes.
>

For logging you should use Platform.getLog(getClass()).warning / error / info -> written from memory, so the API might look at bit different.

**Comment by pzi-dsa** [comment]
> > > error: warnings found and -failOnWarning specified
> > 
> > 
> > I assume that's the problem, because the example (ZipFileStore, to be specific) is using some internal Eclipse classes for "logging" purposes.
> 
> For logging you should use Platform.getLog(getClass()).warning / error / info -> written from memory, so the API might look at bit different.

I probably didn't explain the problem sufficiently. The class uses Policy.error(...) which looks like a normal logging method, but works like:
`if error, then throw CoreException`
Thinking about it more, I can replace the call by throwing the CoreException directly, which would remove at least that dependency.

There's also the problem of reusing the internal error messages stored in the 'org.eclipse.core.internal.filesystem' package. I've looked at how it is handled in other classes of the example, and there the reference has been replaced by a plain string. So I think I'll adapt that solution.

**Comment by vogella** [comment]
I would avoid throwing exceptions if possible, better only to log. But as you said, you can always throw an exception if desired.

**Comment by pzi-dsa** [comment]
> I would avoid throwing exceptions if possible, better only to log. But as you said, you can always throw an exception if desired.

In general, I'd agree. But I'd also like the modifications to the code to a minimum. The code right now is implemented with exceptions in mind. e.g. If throwing a CoreException is replaced with simple logging, then the project would often fail in the following line due to a NullPointerException. Hence why I keep them in there, for now.

I have removed the remaining warnings and the project is compiling. However, the build is instead failing when executing the tycho-eclipserun-plugin with the error:
> Status ERROR: org.eclipse.pde.core code=0 Problems occurred while resolving the target contents children=[Status ERROR: org.eclipse.pde.core code=0 Problems loading repositories children=[Status ERROR: org.eclipse.pde.core code=0 Problems loading repositories children=[Status ERROR: org.eclipse.pde.core code=0 Unable to locate installable unit org.eclipse.ui.examples.filesystem 0.0.0]]]

This seems to be caused when the api-check profile is active. More specifically, it constructs a target-definition containing only the current artifact, which fails, since the artifact hasn't been released yet:
```
<target name="${project.artifactId}-apiBaseline" sequenceNumber="1">
	<locations>
		<location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="planner" includeSource="false" type="InstallableUnit">
			<repository location="https://download.eclipse.org/eclipse/updates/4.23/R-4.23-202203080310/"/>
			<unit id="org.eclipse.ui.examples.filesystem" version="0.0.0"/>
		</location>
	</locations>
</target>
```
I'm not exactly sure what the intent behind this step is, but I assume it is to check, whether the Eclipse can be started with this plugin? What would be the correct approach to solve this problem?

**Review by vogella — APPROVED** [review]

**Comment by vogella** [comment]
Thanks @pzi-dsa for moving and updating this example code.

As this is "just" an example I merge it, as example code does not affected M3.

**Pull request merged**